plugins/autogems.cpp

```diff
@@ -51,6 +57,44 @@ const char *usage = (
     "to cut any accessible rough gems.\n"
 );
 
+static const int32_t persist_version=1;
+
+static void save_config(color_ostream& out);
+static void on_presave_callback(color_ostream& out, void* nothing) {
+    save_config(out);
+}
+
+static void load_config(color_ostream& out) {
+    //if ( !World::isFortressMode() )
+    //    return;
+    Json::Value& p = Persistent::get("autogems");
+    int32_t version = p["version"].isInt() ? p["version"].asInt() : 0;
+    if ( version == 0 ) {
+        //load from histfigs
+        PersistentDataItem config = World::GetPersistentData(CONFIG_KEY);
+        running = config.isValid() && !config.ival(0);
+        last_frame_count = world->frame_counter;
+        World::DeletePersistentData(config);
+    } else if ( version == 1 ) {
+        //load from property_tree
+        running = p["running"].isBool() ? p["running"].asBool() : false;
+    } else {
+        std::cerr << __FILE__ << ":" << __LINE__ << ": unknown version: " << version << endl;
+        exit(1);
+    }
+    running = running && World::isFortressMode();
+    if ( running )
+        plugin_self->plugin_enable(out,true);
+    Persistent::erase("autogems");
```

Comment:
I wanted to do it the same way for different plugins for instructional purposes. If you have a lot of persistent memory you should really be keeping it in binary format instead of text / json.